engine/class_modules/paladin/sc_paladin.hpp

```diff
@@ -937,9 +946,18 @@ struct paladin_action_t : public Base
       am *= 1.0 + p() -> spells.divine_purpose_buff -> effectN( 2 ).percent();
     }
 
+    if ( affected_by.seal_of_reprisal && p()->talents.seal_of_reprisal->ok() )
+    {
+      am *= 1.0 + p()->talents.seal_of_reprisal->effectN( 1 ).percent();
+    }
     if ( affected_by.blessing_of_dawn && p() -> buffs.blessing_of_dawn -> up() )
     {
-      am *= 1.0 + p() -> legendary.of_dusk_and_dawn -> effectN ( 1 ).trigger() -> effectN ( 1 ).percent();
+      am *= 1.0 + p() -> talents.of_dusk_and_dawn -> effectN ( 1 ).trigger() -> effectN ( 1 ).percent();
+    }
+
+    if ( affected_by.divine_purpose && p()->buffs.blessing_of_dawn->up() && p()->talents.seal_of_order->ok() )
+    {
+      am *= 1.0 + p()->talents.seal_of_order->effectN( 1 ).percent();
     }
```

Comment:
we can refactor this to only need to check `buffs.blessing_of_dawn -> up` once, which is marginally better